### DISCUSSION TITLE
Failing to use custom regex for container image registry in Helm chart

### DISCUSSION THREAD

**Discussion by pdonorio** [renovatebot/renovate#27001] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Hello folks, thanks for the great work here.

I have a question: we have helm charts where we used in legacy
versions that we can't update anymore, different naming for
docker repositories than the ones expected by renovate.

For this we are trying to build a custom regex,
but it fails to pick up the registry URL.

As such I've tried to build a minimal reproduction in this public repo:
https://github.com/pdonorio/renovate-test

Latest example job run in the repo:
https://github.com/pdonorio/renovate-test/actions/runs/7748250067/job/21130511560

What is being done incorrectly? Should we maybe use
https://docs.renovatebot.com/modules/datasource/custom/#html-page ? 
May you please point to how in case?

### Logs (if relevant)

<details><summary>Unable to use the URL in regex for registryUrl</summary>

```
WARN: Invalid regex manager registryUrl (repository=pdonorio/renovate-test)
       "value": "ghcr.io"
```

</details>

**Comment by secustor** [comment]
A registryUrl has to be an actual URL rather than a hostname 

Also your versioning will not work there as you have an unsual value there for Docker images. You will need a regex versioning scheme https://docs.renovatebot.com/modules/versioning/#using-a-custom-regex-versioning-scheme

**Reply by pdonorio** [reply]
Hey @secustor appreciate the quick reply 🙏 

> A registryUrl has to be an actual URL rather than a hostname

I'm not entirely familiar with how renovates solves these things, but can we use a registryUrlTemplate to add the https:// in front?

> your versioning will not work 

Thanks for letting me know! We will try to fix the versioning scheme and share this upstream for the image owner. Is there a reference somewhere I could use to suggest the actual versioning expected? 🤔

**Reply by secustor** [reply, marked as answer]
The easiest way for you would be to not using `registryUrl` at all, but create a combined `depName`.

I have not tested the regex so consider this only as pseudo code.
You can use named capture groups of matchStrings inside of templates.
```json
{
   "customManagers": [
    {
       "matchStrings": [
          "repository:\\s+\"?(?<repo>\\S+)\"?\\s+name:\\s+\"?(?<name>\\S+)\"?\\s+tag:\\s+\"?(?<currentValue>\\S+)\""
       ],
       "depNameTemplate": "{{repo}}/{{name}}"
    }
  ]
} 
```

> Thanks for letting me know! We will try to fix the versioning scheme and share this upstream for the image owner. Is there a reference somewhere I could use to suggest the actual versioning expected? 🤔

What is in the current file is will work, but with the namespace concept in your comments is not compatible with the versioning. 
https://docs.renovatebot.com/modules/versioning/#docker-versioning

**Reply by pdonorio** [reply]
Thanks! That's much better!!

regex:
https://github.com/pdonorio/renovate-test/blob/main/renovate.json5#L28

run:
https://github.com/pdonorio/renovate-test/actions/runs/7783175576/job/21221079027

pull:
https://github.com/pdonorio/renovate-test/pull/4
